packages/terra-paginator/src/ProgressivePaginator.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import IconNext from 'terra-icon/lib/icon/IconNext';
+import IconPrevious from 'terra-icon/lib/icon/IconPrevious';
+import ResponsiveElement from 'terra-responsive-element';
+
+import 'terra-base/lib/baseStyles';
+import styles from './Paginator.scss';
+
+import { calculatePages } from './_paginationUtils';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to be executed when a navigation element is selected.
+   */
+  onPageChange: PropTypes.func.isRequired,
+  /**
+   * The active/selected page.
+   */
+  selectedPage: PropTypes.number.isRequired,
+  /**
+   * Total number of all items being paginated.
+   */
+  totalCount: PropTypes.number.isRequired,
+  /**
+   * Total number of items per page.
+   */
+  itemCountPerPage: PropTypes.number.isRequired,
+};
+
+class ProgressivePaginator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { selectedPage, totalCount } = this.props;
+
+    this.state = {
+      pageInput: selectedPage,
+      selectedPage: selectedPage && totalCount ? selectedPage : undefined,
+    };
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+    this.handlePageInput = this.handlePageInput.bind(this);
+    this.defaultProgressivePaginator = this.defaultProgressivePaginator.bind(this);
+    this.reducedProgressivePaginator = this.reducedProgressivePaginator.bind(this);
+  }
+
+  handlePageInput(e) {
+    this.setState({ pageInput: e.target.value });
+  }
+
+  handlePageChange(index) {
+    return (event) => {
+      event.preventDefault();
+
+      this.props.onPageChange(index);
+      this.setState({ pageInput: index, selectedPage: index });
+    };
+  }
+
+  defaultProgressivePaginator() {
+    const totalPages = calculatePages(this.props.totalCount, this.props.itemCountPerPage);
+    const { selectedPage } = this.state;
+
+    return (<div className={cx(['paginator', 'progressive'])}>
+      <div>
+        Page {selectedPage} of {totalPages}
```

Comment:
Should this be internationalized?

Reply:
If it did, we should've called that out earlier on. As it stands, we can re-evaluate as an enhancement.

Reply:
@saedar Can you log a GitHub issue for adding translations to this so we don't forget.

Reply:
Done: https://github.com/cerner/terra-core/issues/1493